presto-main/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -0,0 +1,196 @@
+
+        for (Map.Entry<String, TableColumn> columnMapping : originalColumnMappings.entrySet()) {
+            String viewColumn = columnMapping.getKey();
+            TableColumn originalBaseColumn = columnMapping.getValue();
+
+            Map<SchemaTableName, String> fullBaseColumns = new HashMap<>();
+
+            mappedBaseColumns.forEach(mappedBaseColumnPair -> {
+                if (originalBaseColumn.equals(mappedBaseColumnPair.get(0))) {
+                    fullBaseColumns.put(mappedBaseColumnPair.get(1).getTableName(), mappedBaseColumnPair.get(1).getColumnName());
+                }
+                else if (originalBaseColumn.equals(mappedBaseColumnPair.get(1))) {
+                    fullBaseColumns.put(mappedBaseColumnPair.get(0).getTableName(), mappedBaseColumnPair.get(0).getColumnName());
+                }
+            });
+
+            fullBaseColumns.put(originalBaseColumn.getTableName(), originalBaseColumn.getColumnName());
+
+            fullColumnMappings.put(viewColumn, ImmutableMap.copyOf(fullBaseColumns));
+        }
+
+        return fullColumnMappings.build();
+    }
+
+    public static Session buildOwnerSession(Session session, Optional<String> owner, SessionPropertyManager sessionPropertyManager, String catalog, String schema)
+    {
+        Identity identity = getOwnerIdentity(owner, session);
+
+        return Session.builder(sessionPropertyManager)
+                .setQueryId(session.getQueryId())
+                .setTransactionId(session.getTransactionId().orElse(null))
+                .setIdentity(identity)
+                .setSource(session.getSource().orElse(null))
+                .setCatalog(catalog)
+                .setSchema(schema)
+                .setTimeZoneKey(session.getTimeZoneKey())
+                .setLocale(session.getLocale())
+                .setRemoteUserAddress(session.getRemoteUserAddress().orElse(null))
+                .setUserAgent(session.getUserAgent().orElse(null))
+                .setClientInfo(session.getClientInfo().orElse(null))
+                .setStartTime(session.getStartTime())
+                .build();
+    }
+
+    public static Identity getOwnerIdentity(Optional<String> owner, Session session)
+    {
+        if (owner.isPresent() && !owner.get().equals(session.getIdentity().getUser())) {
+            return new Identity(owner.get(), Optional.empty());
+        }
+        return session.getIdentity();
+    }
+
+    public static Map<SchemaTableName, Expression> generateBaseTablePredicates(Map<SchemaTableName, MaterializedViewStatus.MaterializedDataPredicates> predicatesFromBaseTables, Metadata metadata)
+    {
+        Map<SchemaTableName, Expression> baseTablePredicates = new HashMap<>();
+
+        for (SchemaTableName baseTable : predicatesFromBaseTables.keySet()) {
+            MaterializedViewStatus.MaterializedDataPredicates predicatesInfo = predicatesFromBaseTables.get(baseTable);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.